### PR TITLE
fixing the unit test race condition

### PR DESCRIPTION
**Pull request by toli**
put in a mock for `_send_upstream` to avoid the queue being drained

### CODE REVIEW

File: test/test_producer.py [hunk 1]
```diff
@@ -57,6 +58,8 @@ def test_producer_async_queue_overfilled_batch_send(self):
             message_list = [message] * (queue_size + 1)
             producer.send_messages(topic, partition, *message_list)
         self.assertEqual(producer.queue.qsize(), queue_size)
+        for _ in xrange(producer.queue.qsize()):
+            producer.queue.get()
```
Comment (toli) [inline test/test_producer.py:62]: this is needed to actually drain the queue and avoid a hanging test b/c the async thread never finishes

**Comment by marcusmartins** [comment]
cc @rogaha 

LGTM

File: test/test_producer.py (hunk 1 above)
Comment (rogaha) [inline test/test_producer.py:62]: yeap!

**Comment by rogaha** [comment]
LGTM, I did the same thing locally! :)

**Pull request merged**